src/Microsoft.OData.Core/UriParser/Binders/SelectExpandBinder.cs

```diff
@@ -319,6 +323,15 @@ private SelectItem GenerateSelectItem(SelectTermToken tokenIn)
             parsedPath.AddRange(selectedPath);
             SelectExpandClause selectExpand = BindSelectExpand(null, tokenIn.SelectOption, parsedPath, this.ResourcePathNavigationSource, targetNavigationSource, elementTypeReference, generatedProperties);
 
+            if (isCount)
+            {
+                // Do we need to throw exception if orderBy and nested select expand are not null within /$count ?

```

Comment:
Yes; here, or somewhere, we should return an error if there is anything other than $filter/$search nested in a /$count(...).  [https://github.com/oasis-tcs/odata-abnf/blob/e0b11964a13bf6ebe614b67933da8af0c155de4a/abnf/odata-abnf-construction-rules.txt#L313C2-L313C5](https://github.com/oasis-tcs/odata-abnf/blob/e0b11964a13bf6ebe614b67933da8af0c155de4a/abnf/odata-abnf-construction-rules.txt#L313C2-L313C5)

Reply:
I checked the existing codes as $ref, It just lets them go. I will keep it consistent and we can change them together later if we do need to update.